src/components/FeaturedEventsFilter.tsx

```diff
@@ -0,0 +1,464 @@
+import Calender from '@assets/new/icons/Calender';
+import CancelIcon from '@assets/new/icons/CancelIcon';
+import Location from '@assets/new/icons/Location';
+import { TrainingSupbaseDataType } from '@data/types/trainingTypes/TypeOrthoexTrainingData';
+import { useRouter } from 'next/router';
+import React, { ReactNode, SetStateAction, useCallback } from 'react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import styled from 'styled-components';
+import { DateType, FilterListType } from './FeaturedEvents';
+
+type InputEventType = React.ChangeEvent<HTMLInputElement>;
+
+type CheckboxGroupType = {
+	options: string[],
+	selectedOptions: string[],
+	inputName: string,
+	onChange: (option: string, event: any) => void,
+	children: ReactNode
+}
+
+const CheckboxGroup: React.FC<CheckboxGroupType> = ({
+	options,
+	selectedOptions,
+	onChange,
+	children,
+	inputName,
+}) => {
+	return (
+		<CheckBoxGroupWrapper>
+			<div className="checkbox-title">{children}</div>
+			<div className="options">
+				{options.map((option: string, idx) => (
+					<label key={option}>
+						<input
+							name={`${inputName}-${idx+1}`}
+							type="checkbox"
+							value={option}
+							checked={selectedOptions.includes(option)}
+							onChange={event => onChange(option, event)}
+						/>
+						<span>{option}</span>
+					</label>
+				))}
+			</div>
+		</CheckBoxGroupWrapper>
+	);
+};
+
+function getUniqueValuesByKey<T>(data: T[], key: keyof T): Array<T[keyof T]> {
+	const uniqueValuesSet = new Set<T[keyof T]>();
+
+	data.forEach(item => {
+		uniqueValuesSet.add(item[key]);
+	});
+
+	return Array.from(uniqueValuesSet);
+}
+
+interface FeaturedEventsFilter {
+	filterList: FilterListType;
+	setFilterList: React.Dispatch<SetStateAction<FilterListType>>;
+	training: TrainingSupbaseDataType[];
+}
+
+const FeaturedEventsFilter: React.FC<FeaturedEventsFilter> = ({
+	filterList,
+	setFilterList,
+	training,
+}) => {
+	const locationOptions = getUniqueValuesByKey<TrainingSupbaseDataType>(
+		training,
+		'location',
+	) as string[];
+	const titleOptions = getUniqueValuesByKey<TrainingSupbaseDataType>(
+		training,
+		'title',
+	) as string[];
+
+
+	const router = useRouter();
+
+	const handleQueryParamsOnChange = useCallback(
```

Comment:
why did you choose to use `useCallback` here and below? Just asking.

Reply:
For it to return a optimised function, rather than the normal function. This way it would not be calling it multiple times unnecessarily in the background.

Reply:
Okay, it's basically just because of the dependence on the router. To prevent unnecessary **recreation** of the functions when the router object hasn't changed. Well, I usually just create the function outside and pass the router object as a parameter to the function.